docs/proposals/docker-registry-2-4.md

```diff
@@ -0,0 +1,57 @@
+# Support for Docker Registry 2.4
+## Abstract
+
+A proposal to update the docker/distribution to v2.4.0.
+
+## Motivation
+
+Primary goal is to migrate to Schema 2 [1].
+
+## Compatibility
+
+docker/distribution maintains compatibility with old clients that don't know
+how to work with Schema 2.
+
+##### Docker 1.9
+
+When the manifest is pulled by digest or tag with any docker version, a Schema 1 manifest will be returned.
+
+##### Docker 1.10
+
+Docker Engine 1.10 tries to construct a manifest in the new format when pushing an image.
+If uploading this manifest fails, presumably because the registry only supports the old format,
+Docker will fall back to uploading a manifest in the old format.
```

Comment:
This sounds similar to the authentication pieces where docker tries an insecure connection then, when it fails, it sends credentials.  Does this cause extra logging in the registry as errors?  It has been brought up a couple times that seeing the auth errors concerns users.  Not that I'm suggesting we try to suppress them if it does, but it might be something we need to document.